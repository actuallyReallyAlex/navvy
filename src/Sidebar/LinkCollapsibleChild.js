import React from 'react'
import PropTypes from 'prop-types'
import { Box, Button, Text } from 'grommet'
import LinkIcon from './LinkIcon'

const LinkCollapsibleChild = ({ handler, icon, size, title }) => {
  const handleClick = () => {
    handler && handler()
  }

  return (
    <Button hoverIndicator onClick={handleClick} plain>
      <Box
        background="light-5"
        direction="row"
        height="45px"
        margin={{ vertical: 'none', left: 'medium' }}
        pad={{ vertical: 'none', left: 'small' }}
        style={{ position: 'relative' }}
        width="88%"
      >
        <Box
          align="center"
          direction="row"
          gap="small"
          justify="start"
          width="100%"
        >
          <LinkIcon icon={icon} />
          <Text>{title}</Text>
        </Box>
      </Box>
    </Button>
  )
}

LinkCollapsibleChild.propTypes = {
  handler: PropTypes.func,
  icon: PropTypes.string,
  size: PropTypes.oneOf(['expanded', 'collapsed']).isRequired,
  title: PropTypes.string.isRequired
}

export default LinkCollapsibleChild
