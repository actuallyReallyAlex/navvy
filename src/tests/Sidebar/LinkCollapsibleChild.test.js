import React from 'react'
import LinkCollapsibleChild from '../../Sidebar/LinkCollapsibleChild'
import { shallow } from 'enzyme'

const defaultProps = {
  handler: jest.fn(),
  icon: 'Cube',
  size: 'expanded',
  title: 'Title'
}

describe('<LinkCollapsibleChild />', () => {
  afterEach(() => {
    defaultProps.handler.mockClear()
  })

  test('Should render the <LinkCollapsibleChild /> component.', () => {
    const component = shallow(<LinkCollapsibleChild {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
