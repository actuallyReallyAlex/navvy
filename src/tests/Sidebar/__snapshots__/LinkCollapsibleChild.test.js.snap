// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkCollapsibleChild /> Should render the <LinkCollapsibleChild /> component. 1`] = `ShallowWrapper {}`;
